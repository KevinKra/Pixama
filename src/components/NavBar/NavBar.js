import React from "react";
import "./NavBar.scss";

export default function NavBar() {
  return (
    <nav className="NavBar">
      <p>NavBar</p>
    </nav>
  );
}
