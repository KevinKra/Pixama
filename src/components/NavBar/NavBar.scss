.NavBar {
  border: 1px solid red;
  width: 100%;
  position: fixed;
  z-index: 2;
  background-color: white;
}
