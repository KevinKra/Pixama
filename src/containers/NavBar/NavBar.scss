@import "../../_styles//variables.scss";

.NavBar {
  display: flex;
  justify-content: center;
  border: 1px solid red;
  width: 100%;
  position: fixed;
  // background-color: white;
  // background-color: transparent;
  height: 60px;
  z-index: 5;
  div {
    display: flex;
    justify-content: space-between;
    border: 1px solid red;
    width: 90%;
  }
}

.solid {
  color: $text-dark;
  background-color: white;
  transition: background-color 0.5s ease 0s;
  box-shadow: 0 0 4px grey;
  .nav-links {
    background-color: $text-dark;
    height: 20px;
  }
  a {
    color: $text-light;
  }
}

.transparent {
  color: $text-light;
  background-color: rgba(255, 255, 255, 0);
  transition: background-color 0.5s ease 0s;
  a {
    color: $text-light;
  }
}
