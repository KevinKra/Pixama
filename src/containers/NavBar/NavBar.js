import React, { Component } from "react";
import "./NavBar.scss";
import * as apiCalls from "../../api/apiCalls";
import { connect } from "react-redux";
import { NavLink } from "react-router-dom";
import { bindActionCreators } from "redux";
import { logoutUser, addPopularMovies, addRomanceMovies } from "../../actions";

export class NavBar extends Component {
  state = {
    opacity: false
  };

  componentDidMount() {
    window.addEventListener("scroll", () => {
      let supportPageOffset = window.pageXOffset !== undefined;
      let isCSS1Compat = (document.compatMode || "") === "CSS1Compat";
      let scroll = {
        y: supportPageOffset
          ? window.pageYOffset
          : isCSS1Compat
          ? document.documentElement.scrollTop
          : document.body.scrollTop
      };
      scroll.y > 5
        ? this.setState({ opacity: true })
        : this.setState({ opacity: false });
    });
  }

  handleClick = async () => {
    this.props.logoutUser();
    const popularMovies = await apiCalls.fetchMovies("");
    const romanceMovies = await apiCalls.fetchMovies(
      "&language=en-US&sort_by=popularity.desc&include_adult=false&include_video=false&page=1&with_genres=10749"
    );
    Promise.all([popularMovies, romanceMovies]).then(allMovies => {
      allMovies.forEach((genre, index) => {
        const cleanedGenre = genre.map(movie => {
          return { ...movie, isFavorite: false };
        });
        switch (index) {
          case 0:
            this.props.addPopularMovies(cleanedGenre);
            break;
          case 1:
            this.props.addRomanceMovies(cleanedGenre);
            break;
          default:
            return null;
        }
      });
    });
  };

  render() {
    const { loggedIn } = this.props;
    const returnToMain = () => {
      console.log(this.props.location.pathname);
      if (this.props.location.pathname === "/moviepage")
        return <NavLink to="/mainpage">Main</NavLink>;
    };
    return (
      <nav className={`NavBar ${this.state.opacity ? "solid" : "transparent"}`}>
        <div>
          <h3>PIXAMA</h3>
          {this.props.location.pathname === "/moviepage" && (
            <NavLink to="/">Main</NavLink>
          )}
          {loggedIn ? (
            <p onClick={this.handleClick}>Logout</p>
          ) : (
            <NavLink to="/login">Login</NavLink>
          )}
        </div>
      </nav>
    );
  }
}

export const mapStateToProps = state => ({
  loggedIn: state.currentUser.loggedIn
});

export const mapDispatchToProps = dispatch =>
  bindActionCreators(
    { addPopularMovies, addRomanceMovies, logoutUser },
    dispatch
  );

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(NavBar);
