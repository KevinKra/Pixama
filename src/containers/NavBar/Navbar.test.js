import React from "react";
import { NavBar } from "./NavBar";
import { shallow } from "enzyme";

describe('NavBar', () => {
  let wrapper;
  let instance;
  

  it("should match a snapshot", () => {
    wrapper = shallow(
      <NavBar loggedIn={false} location={{ pathname: "/mainpage" }} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should have default state of opacity false', () => {
    wrapper = shallow(
      <NavBar loggedIn={false} location={{ pathname: "/mainpage" }} />
    );
    expect(wrapper.state('opacity')).toEqual(false);
  });

  it ('should have a property of loggedIn as false', () => {
    wrapper = shallow(
      <NavBar loggedIn={false} location={{ pathname: "/mainpage" }} />
    );
    instance = wrapper.instance();

    expect(instance.props.loggedIn).toEqual(false);
  });

  it.skip('should call addEventListener with scroll on mount', () => {
    const mockListener = { addEventListener: jest.fn() };
    jest
      .spyOn(document, "addEventListener")
      .mockImplementation(() => mockListener);
    wrapper = shallow(
      <NavBar loggedIn={false} location={{ pathname: "/mainpage" }} />
    );
    instance = wrapper.instance();

    expect(
      window.addEventListener("scroll", mockListener)
    ).toHaveBeenCalled();
  });

  it.skip('should call handleClick on click', async () => {
    wrapper = shallow(
      <NavBar loggedIn={true} location={{ pathname: "/mainpage" }} />
    );
    instance = wrapper.instance();

    instance.handleClick = jest.fn();

    wrapper.find('.logout').simulate('click');
    expect(instance.handleClick).toHaveBeenCalled();
  });

  it("should call logoutUser on click", () => {
    wrapper = shallow(
      <NavBar loggedIn={true} location={{ pathname: "/mainpage" }} logoutUser={jest.fn()} />
    );
    instance = wrapper.instance();

    wrapper.find(".logout").simulate("click");
    expect(instance.props.logoutUser).toHaveBeenCalled();
  });
  
});
