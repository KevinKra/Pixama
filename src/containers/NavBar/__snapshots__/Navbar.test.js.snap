// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should match a snapshot 1`] = `
<nav
  className="NavBar transparent"
>
  <div>
    <h3>
      PIXAMA
    </h3>
    <div
      className="routes"
    >
      <NavLink
        to="/login"
      >
        Login
      </NavLink>
    </div>
  </div>
</nav>
`;
